File: Programming file/All programs/Codeforces/C_Heavy_Intervals.cpp
// accepted
// https://codeforces.com/contest/1909/problem/C

#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define ull unsigned long long
#define vsort(a) sort(a.begin(), a.end())
#define vresort(a) sort(a.rbegin(), a.rend())
#define vmin(a) *min_element(a.begin(), a.end())
#define vmax(a) *max_element(a.begin(), a.end())
#define lower_case(a) transform(a.begin(), a.end(), a.begin(), ::tolower)
#define upper_case(a) transform(a.begin(), a.end(), a.begin(), ::toupper)
#define srev(s) reverse(s.begin(), s.end())
#define smax(a) *(a.rbegin()) // 1st check this condition --> if (!a.empty())
#define smin(a) *a.begin()    // 1st check this condition --> if (!a.empty())
const double pi = 3.141592653589793238462643383279502884197169399375105820974944;
const ll N = 1e9 + 7;
// priority_queue<ll>a;
// string s = "dog:cat";  int pos = s.find(":");  string sub = s.substr(pos + 1);   // Copy substring after pos; -->  String is: cat
// string s1 = "Geeks";   string r = s1.substr(3, 2);   // Copy two characters of s1 (starting from position 3); -->  String is: ks

void solve()
{
    int n;
    cin >> n;

    vector<pair<ll, ll>> a;
    vector<ll> c(n);

    for (int i = 0; i < n; i++)
    {
        ll num;
        cin >> num;
        a.push_back({num, 0});
    }
    for (int i = 0; i < n; i++)
    {
        ll num;
        cin >> num;
        a.push_back({num, 1});
    }
    for (auto &num : c)
    {
        cin >> num;
    }

    vsort(a);
    vresort(c);

    stack<ll> stk;
    vector<ll> store;
    for (auto &num : a)
    {
        if (num.second == 0)
        {
            stk.push(num.first);
        }
        else
        {
            ll number = stk.top();
            store.push_back(abs(num.first - number));
            stk.pop();
        }
    }

    vsort(store);

    ll ans = 0;
    for (int i = 0; i < n; i++)
    {
        ans += store[i] * c[i];
    }

    cout << ans << endl;
}

int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    // #ifndef ONLINE_JUDGE
    //     freopen("input.txt", "r", stdin);
    //     freopen("output.txt", "w", stdout);
    // #endif

    int t;
    cin >> t;

    while (t--)
    {
        solve();
    }

    return 0;
}

File: Programming file/All programs/Codeforces/C_Heavy_Intervals.java
// accepted
// https://codeforces.com/contest/1909/problem/C

import java.util.*;

public class C_Heavy_Intervals {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);

        int t = sc.nextInt();

        for (int i = 1; i <= t; i++) {
            solve(sc);
        }
    }

    public static void solve(Scanner sc) {
        int n = sc.nextInt();

        long[][] a = new long[2 * n][2];
        long[] c = new long[n];

        for (int i = 0; i < n; i++) {
            a[i][0] = sc.nextLong();
            a[i][1] = 0;
        }

        for (int i = n; i < 2 * n; i++) {
            a[i][0] = sc.nextLong();
            a[i][1] = 1;
        }

        for (int i = 0; i < n; i++) {
            c[i] = sc.nextLong();
        }

        Arrays.sort(c);

        for (int i = 0; i < n / 2; i++) {
            long temp = c[i];
            c[i] = c[n - 1 - i];
            c[n - 1 - i] = temp;
        }

        Arrays.sort(a, Comparator.comparingLong(arr -> arr[0]));

        Stack<Long> stk = new Stack<>();
        Vector<Long> store = new Vector<>();

        for (int i = 0; i < 2 * n; i++) {
            long num_f = a[i][0], num_s = a[i][1];
            if (num_s == 0) {
                stk.push(num_f);
            } else {
                store.add(num_f - stk.peek());
                stk.pop();
            }
        }

        Collections.sort(store);

        Long ans = 0L;
        for (int i = 0; i < n; i++) {
            ans += store.elementAt(i) * c[i];
        }

        System.out.println(ans);
    }
}
